pkg/k8s/portforward.go

```diff
@@ -162,6 +164,37 @@ func (pf *PortForward) Run() error {
 	return fw.ForwardPorts()
 }
 
+// Init creates and runs a port-forward connection.
+// This function blocks until the connection is established, in which case it returns nil.
+func (pf *PortForward) Init() error {
+	log.Debugf("Starting port forward to %s %d:%d", pf.url, pf.localPort, pf.remotePort)
+
+	failure := make(chan error, 1)
+
+	go func() {
```

Comment:
To confirm my understanding, the `Init()` method will exit when the connection is ready, but this goroutine will block (because `portforward.ForwardPorts()` is [blocking](https://github.com/kubernetes/client-go/blob/2c6e35a5b9cf292580c2d8ecb7d141572997d3cd/tools/portforward/portforward.go#L230-L234)) and live on until a message is sent to `pf.stopCh`, right? Unless the callers call `defer pf.Stop()` (panic etc.), I think this can cause goroutine leaks.

Reply:
Yeah I think `Init()` should be documented that it's the caller's responsibility to call `Stop()`. In `cli/cmd/metrics.go`, this should be a trivial change. `controller/api/public/client.go` looks more involved, as there's an expectation that the portforward continues running after `NewExternalClient` returns. Since `NewExternalClient` is only ever called from the CLI, I'm less concerned about a resource leak, and would be ok deferring that change for a subsequent PR.

Reply:
I'll add `Stop()` to metrics.go.
According to @ihcsim explanation, client.go was already leaking. To fix this we'll have to surface something from portforward into `RunChecks()` that can signal the end of the check in order to close the channel. Agreed can be done in a separate PR.